nodejs/week3/homework-template/src/backend/api/meals.js

```diff
@@ -0,0 +1,92 @@
+const express = require("express");
+const app = express();
+const router = express.Router();
+const knex = require("../database");
+const { returning } = require("../database");
+
+//Returns all meals
+//Get meals that has a price smaller than maxPrice
+//Get meals that still has available reservations
+//Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+//Get meals that has been created after the date
+//Only specific number of meals
+router.get("/", async (request, response) => {
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    if (request.query.maxPrice) {
+      const maxPrice = Number(request.query.maxPrice);
+      const mealsWithLesserPrice = await knex("meal")
+        .select("*")
+        .where("Price", "<", maxPrice);
+      response.json(mealsWithLesserPrice);
+    } else if (request.query.availableReservations) {
+      const mealsWithAvailableReservation = await knex("meal")
+        .select("title", "description")
+        .join("reservation", "meal.id", "=", "reservation.meal_id")
+        .where("max_reservations", ">", "number_of_guests")
+        .groupBy("reservation.meal_id");
+      // TODO: Fix having clause query later. Need to read knex documentation
+      // .having("Sum(reservation.number_of_guests)","<","meal.max_reservations");
+      response.json(mealsWithAvailableReservation);
+    } else if (request.query.title) {
+      const mealTitle = request.query.title;
+      const mealsWithMatchingTitle = await knex("meal")
+        .select("*")
+        .where("title", "like", `%${mealTitle}%`);
+      response.send(mealsWithMatchingTitle);
+    } else if (request.query.createdAfter) {
+      const createdDate = new Date(request.query.createdAfter);
+      const mealsCreatedAfterDate = await knex("meal")
+        .select("*")
+        .where("created_date", ">", createdDate);
+      response.json(mealsCreatedAfterDate);
+    } else if (request.query.limit) {
+      const numberLimit = Number(request.query.limit);
+      const mealaUptoLimit = await knex("meal").select("*").limit(numberLimit);
+      response.json(mealaUptoLimit);
+    } else {
+      const titles = await knex("meal").select("title");
```

Comment:
I think i would have flipped this on its head. First taking the "standard way". So in the code, first check if there are not queries if not respond with the titles. 
Other wise continue with the queries